src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -568,15 +569,28 @@ FancyZones::OnKeyDown(PKBDLLHOOKSTRUCT info) noexcept
         }
     }
 
-    if (!shift && win && ctrl && alt && m_settings->GetSettings()->quickLayoutSwitch)
+    if (m_settings->GetSettings()->quickLayoutSwitch)
     {
+        int digitPressed = -1;
         if ('0' <= info->vkCode && info->vkCode <= '9')
+        {
+            digitPressed = info->vkCode - '0';
+        }
+        else if (VK_NUMPAD0 <= info->vkCode && info->vkCode <= VK_NUMPAD9)
+        {
+            digitPressed = info->vkCode - VK_NUMPAD0;
+        }
+
+        bool dragging = m_windowMoveHandler.IsDragEnabled();

```

Comment:
What about first checking if `dragging` is true and only in that case execute the code from line 575 to 584?
Performance wise the difference is minimal but it makes the code follow a more structured steps:
 - check if `quickLayoutSwitch` is on
 - check if `dragging` is on and a digit was pressed
 - check if a shortcut is pressed

Reply:
You are right.